File: Training/1.1 Dabble/1.1.3-friday.cpp
/*
ID:   andy1li
LANG: C++
TASK: friday
*/

#include <bits/stdc++.h>
using namespace std;
using vi = vector<int>;

#define range(stop) for (size_t i=0; i<stop; ++i)
#define rep(i, start, stop) for (size_t i=start; i<stop; ++i)

/*----------------------------------------------------------------------------*/

int is_leap(int year) {
    return year%4==0 && (year%100 != 0 || year%400 == 0);
}

int days_in_month(int year, int month) {
    vi days = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    return days[month] + (month == 1 && is_leap(year));
}

void solve() {
    int n; cin >> n;

    vi count(7, 0);
    int day_of_week = 0; // day of week: January 13, 1900 was a Saturday = 0

    rep(year, 1900, 1900+n) {
        rep(month, 0, 12) {
            count[day_of_week]++;
            day_of_week += days_in_month(year, month)
            day_of_week %= 7
        }
    }

    range(7) { 
        cout << count[i] << ((i < 6) ? ' ' : '\n'); 
    }
}

/*----------------------------------------------------------------------------*/

int main() {
    freopen("friday.in",  "r", stdin);
    freopen("friday.out", "w", stdout);
    solve();
}

File: Training/1.1 Dabble/1.1.3-friday.py
from __future__  import print_function

def is_leap(year): return year%4==0 and (year%100 != 0 or year%400 == 0)

def days_in_month(year, month):
    if   month == 2            : return 28 + is_leap(year)
    elif month in [4, 6, 9, 11]: return 30
    else                       : return 31

with open('friday.in') as fin:
    n  = int(fin.readline())

    thirteenths = [0] * 7
    day_of_week  = 0 # day of week: January 13, 1900 was a Saturday = 0 
    for year in range(1900, 1900+n):
        for month in range(1, 13):
            thirteenths[day_of_week] += 1
            day_of_week = (day_of_week + days_in_month(year, month)) % 7

    print(*(thirteenths[day] for day in range(7)))
